### ISSUE TITLE
Add AA Openness Index (AAOI) scores to AAII-scored models + home-page column

### ISSUE THREAD

**Issue by hammer** [hammer/labs#42]
Surface [Artificial Analysis's Openness Index](https://artificialanalysis.ai/evaluations/artificial-analysis-openness-index) on labindex.ai alongside the existing AA Intelligence Index. Two deliverables: (1) backfill an \`openness_index\` field on every model output that already has \`intelligence_index\`, and (2) add an "Openness" column to the home-page table showing the AAOI of each lab's top-AAII-scoring model.

## What AAOI is

AA's AAOI is a **0–100 normalized score** of how open a model is. Underlying it is an 18-point qualitative rubric across two dimensions:

- **Model Availability (6 pts):** weights access (0–3) + commercial licensing (0–3)
- **Model Transparency (12 pts):** pre-training data access (0–3), pre-training data license (0–3), post-training data access (0–3), post-training data license (0–3), methodology disclosure (0–3), code licensing (0–3) — data items averaged to 6 pts total, so total caps at 18

Spec PDF: [/downloads/AA-Openness-Index-Spec_V1-0.pdf](https://artificialanalysis.ai/downloads/AA-Openness-Index-Spec_V1-0.pdf). Current version is **AA Openness Index v1.0**. Scoring is **per-model checkpoint**, including thinking vs non-thinking variants getting distinct entries (e.g. \`Llama Nemotron Super 49B v1.5 (Reasoning)\` and the non-reasoning sibling each get their own score).

234 models scored as of May 16, 2026. Range observed in the wild:

| Score | Example |
|---|---|
| 88.89 (16/18) | OLMo 3 / Apertus / K2-V2 / Molmo |
| 83.33 (15/18) | Nemotron 3 Super 120B A12B |
| 50.00 (9/18) | DeepSeek V4 Pro, Phi-4, Magistral Small 1.2, Gemma 3 27B Instruct |
| 33.33 (6/18) | Kimi K2.6, Hy3-preview |
| 11.11 (2/18) | Claude 4.5 Sonnet/Haiku, Nova variants, Doubao Seed Code |
| 5.56 (1/18) | GPT-5 family, Gemini 2.5, Grok 4, o3 |

## How to gather the data

**No public CSV/JSON API exists.** Probed \`/api/models\`, \`/sitemap-models.xml\`, provider pages, and the leaderboard page directly — none expose the full dataset. The leaderboard page (\`/evaluations/artificial-analysis-openness-index\`) inlines only 16 chart-highlight entries; the rest of the 234 are rendered client-side from an API the page makes after load.

**But each per-model page reliably embeds its own score** in inline JSON:

\`\`\`
"opennessIndex": <value>, "detailsUrl": "/models/<aa-slug>"
\`\`\`

Verified by parsing \`/models/seed-oss-36b-instruct\` (44.44) and \`/models/doubao-seed-code\` (11.11) — both match the leaderboard. So the gathering strategy is:

1. For each labindex output that has \`intelligence_index\`, derive the AA slug from the \`sources\` URL pointing at \`artificialanalysis.ai/models/<slug>\`.
2. Fetch \`https://artificialanalysis.ai/models/<slug>\` (one HTTP request per model).
3. Extract the model's own score with regex anchored on \`"detailsUrl":"/models/<slug>"\`.
4. Record \`openness_index\` and \`openness_index_version: "AA Openness Index v1.0"\` on the model entry.

**Audit of in-scope outputs:** 97 outputs currently have \`intelligence_index\` set. **84** have a parseable AA URL in \`sources\`; the remaining **13** need a manual AA-slug lookup before the script can resolve them:

\`\`\`
data/outputs/ai21/jamba.yaml             (intel=11)
data/outputs/cohere/tiny-aya.yaml        (intel=5)
data/outputs/deepseek/deepseek-v4.yaml   (intel=52)
data/outputs/liquid-ai/lfm2.yaml         (intel=10)
data/outputs/microsoft/phi-4.yaml        (intel=10)
data/outputs/mistral/devstral.yaml       (intel=22)
data/outputs/mistral/magistral.yaml      (intel=27)
data/outputs/mistral/ministral-3.yaml    (intel=16)
data/outputs/moonshot-ai/kimi-k2.6.yaml  (intel=54)
data/outputs/reka/reka-core.yaml         (intel=10)
\`\`\`
*(plus 3 more — full list when implementation lands)*

## Schema changes

Add to \`ModelDetailsSchema\` in \`src/schema.ts\`, next to \`intelligence_index\`:

\`\`\`ts
openness_index: z.number().optional(),
openness_index_version: z.string().optional(),
\`\`\`

Mirror the existing AAII convention — version string tag (\`"AA Openness Index v1.0"\`) sits alongside the score so future spec revisions are easy to spot. Store the raw float (e.g. \`44.4\`); render rounded to one decimal.

## Home-page column: \"Openness of top-AAII model\"

The user-requested rule is: **for each lab, find the model with the highest \`intelligence_index\` and surface its \`openness_index\`**. Notably this is *not* the lab's highest AAOI — it's whichever model is the lab's flagship intelligence-wise. This pairs the openness reading with the same model the Intelligence column links to.

### Loader changes

Extend \`getTopIntelligence()\` in \`src/data/loader.ts\` (lines 222–250) to also surface that model's openness:

\`\`\`ts
export function getTopIntelligence(labSlug: string): {
  score: number;
  openness?: number;            // ← new
  opennessVersion?: string;     // ← new
  name: string;
  slug: string;
  labSlug: string;
} | null
\`\`\`

The \`check\` inner function already iterates per-model. Pass \`sub.model.openness_index\` / \`output.model.openness_index\` through.

### \`src/pages/index.astro\` changes

Five edits, all next to the existing Intelligence column:

1. **Columns array** (line ~70): add
   \`{ key: 'openness', label: 'Openness', kind: 'range', rowAttr: 'openness', unit: '/100', excludeMissing: true, ... }\`
2. **\`<th>\`** (line ~96): \`<th class=\"col-openness sortable\" data-sort=\"openness\">Openness</th>\`
3. **Row data attribute** (line ~121): \`data-openness={intel?.openness ?? 0}\`
4. **\`<td>\`** (line ~155-area): one decimal place, link to the same output as Intelligence, dash if missing
5. **Sort logic** in the \`<script>\` block

Placement: between **Intelligence** and **Outputs** — the natural read order is Scale → Intelligence → Openness → Outputs.

### Color-coding (optional, second pass)

AAOI clusters into bands (~5/11/17/22/28/33/39/44/50/56/61/67/72/78/83/89 — these are the 1/18 through 16/18 integer steps). Could optionally tint cells by band (e.g. green ≥ 67, yellow 33–66, red < 33). Defer to a follow-up; first pass should just print the number.

## Skills + docs updates

- **AGENTS.md** § Artificial Analysis Intelligence Index: add a parallel \"Artificial Analysis Openness Index\" subsection mirroring the existing AAII guidance. Document the v1.0 version tag, the spec PDF, the per-model-page scraping pattern, and the rule that scores are per-checkpoint (so reasoning and non-reasoning variants should each get their own openness_index when both have an AAII entry).
- **\`.agents/skills/add-output/SKILL.md\`**: add an \"AA Openness Index\" callout in the model-details section, paralleling the existing AAII guidance.
- **Memory**: add \`ref_aa_openness_index.md\` analogous to the existing \`ref_aa_intelligence_index.md\`, including the per-model-page scraping pattern (so future me doesn't re-derive it).

## Implementation phases

1. **Schema + loader + page (no data yet).** Land the field, the column, and the loader — all values render as \"—\" until backfill ships. Verify build is green with no data populated.
2. **Backfill script.** One-off \`scripts/fetch-aa-openness.ts\` that walks every output with \`intelligence_index\`, fetches its AA page, parses the score, and writes \`openness_index\` + \`openness_index_version\` back to the YAML. Idempotent (overwrites existing values, since AA can re-score).
3. **Run the backfill** on the 84 outputs that have a parseable AA URL. Commit the YAML diff.
4. **Manual fixup** for the 13 outputs missing AA URLs in \`sources\`. For each, look up the AA slug (or confirm AA doesn't track the model), add the \`Artificial Analysis\` source if appropriate, re-run the script.
5. **Docs + memory** updates land in the same PR as Phase 1.

## Adversarial review

- *\"Why scrape — won't AA add this to their CSV exports eventually?\"* Possible, but no announced timeline. Scrape now; if AA ships an API, switch the script.
- *\"What about models AA hasn't scored on AAOI yet?\"* AAOI coverage (234) is smaller than AAII coverage (357+ on the leaderboard). Models without an AAOI page entry get \`openness_index\` left unset → column renders \"—\".
- *\"Should AAOI be the column-shown number for the *highest-AAOI* model, not the highest-AAII?\"* User explicitly asked for the highest-AAII model's openness. That's the rule. (A follow-up could add a separate \"max openness\" column.)
- *\"Are the version tags worth the cost?\"* Yes. AAII v3.0→v4.0 silently invalidated 10+ point swings; AAOI will follow the same pattern. Version tags let a one-line grep find every score that needs re-verification on a methodology bump.
- *\"Per-model fetch is 84+ requests — rate limits?\"* AA's site is Cloudflare-fronted; in practice 1 req/sec with a polite User-Agent is fine. Script will pace at 500–1000 ms intervals.

## Sanity-check examples (gathered during research)

| labindex output | AAII | AAOI |
|---|---|---|
| \`bytedance-seed/seed-oss\` | 25 | 44.44 |
| \`bytedance-seed/doubao-seed-code\` | 34 | 11.11 |
| \`amazon/nova-premier\` | 19 | 11.11 |

When implementation lands, these are the first three to verify on the live site.

**Comment by hammer** [comment]
## Adversarial review of the plan

Did a deeper probe of AA's data layer and the labindex slug surface area. Five things in the original plan are wrong or under-specified.

### 1. The per-model-page scrape is unreliable

I claimed each \`/models/<slug>\` page reliably embeds its own AAOI inline. **It doesn't.** The openness chart on each model page shows ~9 *comparator* models — only including the page's own model when it's competitive in the chart's score range.

| Model | Self-AAOI inline? |
|---|---|
| seed-oss-36b-instruct (44.44) | ✓ |
| doubao-seed-code (11.11) | ✓ |
| gpt-5-4 (closed) | ✗ |
| claude-opus-4-7 (closed) | ✗ |
| gemini-3-1-pro-preview (closed) | ✗ |

So my proposed scrape would systematically miss closed frontier models — exactly the ones whose openness reading is most useful to surface.

**Better approach:** the AAOI leaderboard page exposes the full dataset via React Server Component when fetched with the \`RSC: 1\` header (or \`?_rsc=1\` query param). One request returns 3.97 MB containing **all 234 records** including \`slug\`, \`opennessIndex\`, and the six component sub-scores (\`dataPretrainAccess\`, \`dataPretrainLicense\`, \`dataPosttrainAccess\`, \`dataPosttrainLicense\`, \`transparencyMethodology\`, \`transparencyPreTrainingData\`, \`transparencyPostTrainingData\`, \`modelAvailability\`, \`modelTransparency\`).

Parse pattern that actually works (verified against the full 234 records):

\`\`\`python
records = re.split(r'(?=\\{\"additional_text\")', rsc)
for r in records:
    slug = re.search(r'\"slug\"\\s*:\\s*\"([^\"]+)\"', r).group(1)
    aaoi = float(re.search(r'\"opennessIndex\"\\s*:\\s*([\\d.]+)', r).group(1))
\`\`\`

One HTTP request, complete coverage, robust to AA's chart layout changes. Phase 2 of the rollout should be rewritten around this. 84+ per-model fetches → 1.

### 2. Coverage is far worse than I estimated

Original plan assumed 84 of 97 labindex outputs would resolve. Actual numbers using the RSC dataset:

- **48 of 84** have a direct slug match in AAOI v1.0 (57%)
- **36 of 84** have an AA URL in \`sources\` but **the specific slug isn't in AAOI**

Examples of the 36 misses:
\`\`\`
claude-opus-4-7        (labindex's slug)  →  AAOI tracks claude-opus-4-5, claude-opus-4-5-thinking
nova-pro / nova-2-0-pro                   →  not in AAOI v1.0
qwen2-5-coder-32b-instruct                →  AAOI tracks Qwen3.x family, not Qwen2.x
deepseek-v3-1, deepseek-v3-2-reasoning   →  AAOI tracks deepseek-v3-1-terminus etc.
ernie-5-0-thinking-preview               →  not in AAOI v1.0
\`\`\`

So Phase 4 \"manual fixup\" isn't 13 outputs — it's closer to **~49**. For some (Claude, Nova, ERNIE 5) the answer is genuinely \"no AAOI exists for this model.\" For others (Qwen2, DeepSeek V3.1) the fix is to *sibling-map* to a related variant. Decide policy up front; don't bury this in \"manual fixup later.\"

### 3. Slug-version mismatch is structural, not a one-time cleanup

AA scores AAOI **per checkpoint** at a particular point in time; labindex tracks **the latest checkpoint** of each family. As labs ship new versions, AAOI catches up over weeks. The 36-miss list isn't a backlog to clear — it's a recurring drift the loader has to handle.

**Decision needed:** when the top-AAII model has no direct AAOI match, what should the home page show?

- **Option A — strict:** render \"—\". Anthropic, Google Gemini, ERNIE 5 will all show blank for now, until AA scores the current checkpoints. *(Predictable; surfaces the gap.)*
- **Option B — family fallback:** if \`claude-opus-4-7\` has no AAOI, fall back to \`claude-opus-4-5-thinking\` (same family). Requires a family-slug map. *(More columns filled; risks attributing wrong score.)*
- **Option C — \"top AAII model with AAOI\":** pick the highest-AAII model that also has an AAOI score, even if it's not THE top-AAII model. *(Doesn't require a family map; loses the \"same model linked from both columns\" property.)*

My recommendation: **Option C**. The column header reads \"Openness\" with an underscored tooltip explaining \"of the highest-AAII labindex model that AA has also scored on AAOI as of <date>.\" Implementation: in \`getTopIntelligence()\`, sort outputs by AAII desc and pick the first one whose \`openness_index\` is also set.

### 4. Same-model variants disagree on AAOI

A model family can have many AAOI entries (Qwen3 has 40+) with **scores varying by ±30 points** depending on variant:

\`\`\`
qwen3-max-thinking-preview          16.67   (proprietary)
qwen3-5-9b                          38.89
qwen3-vl-8b-reasoning               50.00
\`\`\`

AGENTS.md already has the rule \"top-level \`intelligence_index\` should match the model named by the file slug and the AA URL in \`sources\` — not blindly the highest variant.\" **Mirror that rule for openness_index** so labindex's stored AAOI is the one for the *same checkpoint* whose AAII we recorded — not the family's max openness. Add this explicitly to AGENTS.md.

### 5. Score precision

AAOI is always a multiple of 1/18 ≈ 5.555... — so storing \`5.555555555555555\` in YAML adds noise. Either:

- Store rounded to one decimal: \`5.6\` (matches AA's leaderboard display)
- Store the underlying numerator: \`openness_index: 1\` with an \`openness_index_max: 18\` (more honest, less familiar)

Recommendation: store \`5.6\` (rounded float). Display as integer on the column when ≥10 (e.g., \"50\"), one decimal when <10 (e.g., \"5.6\"). Tooltip can expand to the raw fraction.

### Minor

- **Robots.txt:** confirmed AA allows all crawlers (\`User-agent: * / Allow: /\`). Good.
- **Scraping rate:** moot now since we're at 1 request total. Phase 2 is dramatically simpler.
- **Component sub-scores are available too.** The RSC payload exposes \`modelAvailability\`, \`modelTransparency\`, and six 0-3 sub-scores per record. **Not proposing to store these** — the composite is the right primary surface — but they exist if a future per-model \"why\" tooltip wants to show the breakdown.
- **The 13 missing-AA-URL outputs** (jamba, tiny-aya, devstral, magistral, ministral-3, etc.) are mostly genuinely missing from AA's main AAII coverage too. Phase 4 might end up being \"verify these are AA-untracked and document.\"

### Revised plan (TL;DR diff)

- Replace per-model-page scrape with single RSC-payload fetch. Phase 2 becomes a ~50-line script.
- Adopt **Option C** for column population: top-AAII-with-AAOI. Add the rule to AGENTS.md.
- Add **per-checkpoint matching rule** for \`openness_index\` to AGENTS.md, mirroring the existing AAII rule.
- Round AAOI to one decimal in YAML.
- Realistic coverage expectation: **~50–60 outputs gain AAOI on first run**; the remainder are AA-version-drift cases that fill in over time as AA scores newer checkpoints.

Happy to fold this into the issue body or proceed with the revised plan.

**Comment by hammer** [comment]
## Confirming AAOI filtering is in scope

Re-checked the existing implementation. The home-page columns and filters are wired through a single \`dimensions\` array in \`src/pages/index.astro\` (lines 56–74) consumed by \`<FilterBar mode=\"palette\" dimensions={dimensions} hotkey=\"f\" />\` — so the columns-array entry I proposed is what wires up the filter. Calling that out explicitly:

\`\`\`ts
{ key: 'openness',
  label: 'Openness',
  kind: 'range',
  rowAttr: 'openness',
  unit: '/100',
  excludeMissing: true,
  format: (n) => String(n) },
\`\`\`

This gives the same affordances the Intelligence column already has:

| Affordance | Mechanism |
|---|---|
| Range slider in the \`f\` filter palette | \`kind: 'range'\` |
| Skip labs with no AAOI value when slider is active | \`excludeMissing: true\` (state.ts:242 treats \`data-openness=\"0\"\` as missing) |
| URL-serializable filter state | inherited from \`FilterBar\` |
| Sortable column header | the matching \`<th data-sort=\"openness\" class=\"sortable\">\` plus a case in the sort-logic switch |
| Inclusion in keyboard-shortcut filter cycle | dimensions array is the single source of truth |

The home-page row gets the data attribute \`data-openness={intel?.openness ?? 0}\` — which means **\"0\" is the missing-data sentinel**, picked up by \`excludeMissing\`. AAOI's lowest real score is 5.56, so \`0\` never collides with a real value.

**Default sort behavior:** new column has \`data-sort=\"openness\"\`; the existing sort logic in the index.astro \`<script>\` block needs the parallel case (descending = most open first). Same one-line addition the other range columns have.

**Range presentation:** AAOI scores cluster at the 1/18 increments (5.56, 11.11, 16.67, 22.22, ...). The slider default min/max should clamp to the observed range — same auto-clamping the Intelligence slider does (\`format: (n) => String(n)\`). \`unit: '/100'\` puts the right hint in the panel header.

No additional plumbing needed beyond what the original plan listed; the dimensions-array entry handles both filter UI and the column-metadata role.
